src/WebJobs.Script/Rpc/LanguageWorkerConsoleLogService.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    internal class LanguageWorkerConsoleLogService : IHostedService, IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly ILanguageWorkerConsoleLogSource _source;
+        private readonly CancellationTokenSource _cts = new CancellationTokenSource();
+        private Task _processingTask;
+        private bool _disposed = false;
+
+        public LanguageWorkerConsoleLogService(ILoggerFactory loggerFactory, ILanguageWorkerConsoleLogSource consoleLogSource)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _source = consoleLogSource ?? throw new ArgumentNullException(nameof(consoleLogSource));
+            _logger = loggerFactory.CreateLogger(LanguageWorkerConstants.FunctionConsoleLogCategoryName);
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _processingTask = ProcessLogs();
+            return Task.CompletedTask;
+        }
+
+        public async Task StopAsync(CancellationToken cancellationToken)
+        {
+            _cts.Cancel();
+            await _processingTask;
```

Comment:
We don't know when the logging is done so we need to just stop listening at some point. If it's a host restart, any extra logs will be buffered and then logged when the next JobHost starts. If the whole process is shutting down, there's a chance we miss some of those last-second console.log() calls (assuming the process that's writing them lives a little longer than this logger). My suggested "rebuild" of logging would help with that as the loggers would live with the full WebHost rather than with the ScriptHost. But for now, it's hard for us to do any logging outside the scope of a running JobHost, and this call to StopAsync() means the JobHost() is shutting down.